samples/apps/copilot-chat-app/webapi/Skills/ChatSkill.cs

```diff
@@ -212,27 +215,116 @@ public async Task<string> AcquireExternalInformationAsync(SKContext context)
         if (plan.Steps.Count > 0)
         {
             SKContext planContext = await plan.InvokeAsync(plannerContext);
+            int tokenLimit = int.Parse(context["tokenLimit"], new NumberFormatInfo());
 
             // The result of the plan may be from an OpenAPI skill. Attempt to extract JSON from the response.
             if (!this.TryExtractJsonFromOpenApiPlanResult(planContext.Variables.Input, out string planResult))
             {
                 // If not, use result of the plan execution result directly.
                 planResult = planContext.Variables.Input;
             }
+            else
+            {
+                int relatedInformationTokenLimit = (int)Math.Floor(tokenLimit * this._promptSettings.RelatedInformationContextWeight);
```

Comment:
The output still may be JSON - could we remove this from an else clause and make it resilient to non-JSON content?